code/modules/mob/living/carbon/hallucination.dm

```diff
@@ -0,0 +1,1102 @@
+	for(var/mob/living/M in oview(holder))
+		hal_target += M
+	if(hal_target.len)
+		to_chat(holder, "<b>[pick(hal_target)]</b> [t]")
+
+/datum/hallucination/paranoia/second		//Just so we get another chance at picking this.
+
+
+
+
+/datum/hallucination/skitter
+	max_power = 60
+/datum/hallucination/skitter/start()
+	to_chat(holder, "The spiderling skitters around.")
+
+
+
+/datum/hallucination/prick
+	duration = 40
+	max_power = 35
+
+/datum/hallucination/prick/start()
+	to_chat(holder,SPAN_NOTICE("You feel a tiny prick!"))
+
+/datum/hallucination/prick/end()		//chance to feel another effect after duration time
+	switch(rand(1,6))
+		if(1)
+			holder.druggy += min(holder.hallucination, 15)
+		if(2)
+			holder.make_dizzy(105)
+		if(3)
+			to_chat(holder,SPAN_GOOD("You feel good."))
+	..()
+
+/datum/hallucination/prick/by_person	//the prick feeling but you actually imagine someone injecting you
+	min_power = 30
+	max_power = INFINITY
+	duration = 20
+	var/injector
+	var/needle
+	var/list/prick_candidates = list()
+
+/datum/hallucination/prick/by_person/can_affect(mob/living/carbon/C)
+	if(!..())
+		return FALSE
+	for(var/mob/living/M in oview(C, 1))
+		if(!M.stat)
+			prick_candidates += M
+	if(prick_candidates.len)
+		return TRUE
+
+/datum/hallucination/prick/by_person/start()
+	injector = pick(prick_candidates)
+	needle = pick("syringe", "hypospray", "pen")
+	to_chat(holder, SPAN_WARNING("\The [injector] is trying to inject \the [holder] with \the [needle]!"))
+
+/datum/hallucination/prick/by_person/end()
+	to_chat(holder,SPAN_NOTICE("\The [injector] injects \the [holder] with \the [needle]!"))
+	to_chat(holder,SPAN_NOTICE("You feel a tiny prick!"))
+	..()
+
+
+
+/datum/hallucination/insides
+	duration = 60
+	max_power = 75
+
+/datum/hallucination/insides/start()
+	if(ishuman(holder))
+		var/mob/living/carbon/human/H = holder
+		var/obj/O = pick(H.organs)
+		to_chat(holder,SPAN_DANGER("You feel something [pick("moving","squirming","skittering", "writhing", "burrowing", "crawling")] inside of your [O.name]!"))
+	else
+		to_chat(holder,SPAN_DANGER("You feel something [pick("moving","squirming","skittering", "writhing", "burrowing", "crawling")] inside of you!"))
+	if(prob(min(holder.hallucination/2, 80)))
+		sound_to(holder, pick('sound/misc/zapsplat/chitter1.ogg', 'sound/misc/zapsplat/chitter2.ogg', 'sound/effects/squelch1.ogg', 'sound/effects/lingextends.ogg'))
+
+/datum/hallucination/insides/end()
+	if(prob(50))
+		to_chat(holder, SPAN_WARNING(pick("You see something moving under your skin!", "Whatever it is, it's definitely alive!", "If you don't get it out soon...", "It's moving towards your mouth!")))
+	..()
+
+
+
+/datum/hallucination/pain				//Pain. Picks a random type of pain, and severity is based on their level of hallucination.
+/datum/hallucination/pain/start()
```

Comment:
There's also inconsistent spacing all across the file here from the declaration to the first proc.